File: 프로그래머스/2/64065. 튜플/튜플.js
function solution(s) {
    let arr = []
    s = s.slice(1,-1)
    let li = ''
    for (let i of s) {
        if (i === '{') {
            li = '';
        } else if (i === '}'){
            arr.push(li.split(','))
            li = '';
        } else {
            li += i
        }
    }
    
    arr.sort((a,b) => a.length - b.length);
    const answer =[]
    const set_arr = new Set();
    for (let li of arr) {
        for (let num of li) {
            num = parseInt(num)
            if (!set_arr.has(num)) {
                answer.push(num);
                set_arr.add(num)
            }
        }
    }
    return answer;
}

File: 프로그래머스/2/64065. 튜플/튜플.py
def solution(s):
    arr = []
    s=s[1:-1]
    li = ''
    for i in s:
        if i == '{':
            li=''
        elif i == '}':
            arr.append(list(li.split(',')))
            li=''
        else:
            li+=i
    arr.sort(key=lambda x:len(x))
    answer = []
    for li in arr:
        for num in li:
            if int(num) not in answer:
                answer.append(int(num))
    return list(answer)